source/extensions/filters/network/rocketmq_proxy/router/router_impl.cc

```diff
@@ -0,0 +1,237 @@
+#include "extensions/filters/network/rocketmq_proxy/router/router_impl.h"
+
+#include "extensions/filters/network/rocketmq_proxy/active_message.h"
+#include "extensions/filters/network/rocketmq_proxy/codec.h"
+#include "extensions/filters/network/rocketmq_proxy/conn_manager.h"
+#include "extensions/filters/network/rocketmq_proxy/protocol.h"
+#include "extensions/filters/network/rocketmq_proxy/well_known_names.h"
+#include "extensions/filters/network/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RocketmqProxy {
+
+RouterImpl::RouterImpl(Envoy::Upstream::ClusterManager& cluster_manager)
+    : cluster_manager_(cluster_manager), handle_(nullptr), active_message_(nullptr) {}
+
+RouterImpl::RouterImpl(Envoy::Upstream::ClusterManager& cluster_manager,
+                       absl::string_view target_host)
+    : cluster_manager_(cluster_manager), handle_(nullptr), active_message_(nullptr),
+      target_host_(target_host.data(), target_host.size()) {}
+
+void RouterImpl::onAboveWriteBufferHighWatermark() {
+  ENVOY_LOG(trace, "Above write buffer high watermark");
+}
+
+void RouterImpl::onBelowWriteBufferLowWatermark() {
+  ENVOY_LOG(trace, "Below write buffer low watermark");
+}
+
+void RouterImpl::onEvent(Network::ConnectionEvent event) {
+  switch (event) {
+  case Network::ConnectionEvent::RemoteClose: {
+    ENVOY_LOG(error, "Connection to upstream: {} is closed", upstream_host_->address()->asString());
+    active_message_->onError("Connection to upstream is closed");
+    break;
+  }
+  case Network::ConnectionEvent::LocalClose: {
```

Comment:
Here, the client should actively disconnect from upstream instead of downstream

Reply:
Fixed